### DISCUSSION TITLE
Support distinguishing between development vs production dependencies

### DISCUSSION THREAD

**Discussion by leonsilicon** [pkgxdev/pkgx#606] [Ideas]
# Use Case

I have some dependencies that I want tea to install when I'm developing the app locally (e.g. `minikube` for a local kubernetes cluster and `kubectl` to interact with it), but then in CI/CD I only need a subset of those dependencies (I wouldn't need `minikube` in CI/CD but I still need `kubectl` for updating the live kubernetes cluster).

Right now, all these dependencies (including development-only dependencies) are specified in one `tea.yaml` file in the root of my repository and there doesn't seem to be a way to make tea filter out certain dependencies to skip in production, or to install dependencies from a separate YAML file on CI/CD (e.g. a `tea.prod.yaml` file) that I'd specify production-only dependencies in.

**Comment by jhheider** [comment]
This is an interesting idea. One option might be something like this:
```sh
git add tea.dev.yaml tea.prod.yaml
...
git clone foo
ln -s tea.dev.yaml tea.yaml
cd .
```

You lose some of the automaticality[wordmark pending], but you'd get finer-grained control. You could also use a tea-script in the language of your choice to manage that for you. *In theory*, extra deps are harmless but not costless; this isn't always true, but it should be true in your scenario. You could also compose your environments manually, say, by having a file like `setup-dev-env.sh`:
```sh
#!/bin/sh

tea +kubernetes.io/minikube sh
```
To allow loading your richer dev-env.

**Comment by mxcl** [comment]
Good call. I'm surprised this hasn't come up before.

So one file would be:

```yaml
dependencies:
   foo.com: ^1

production:
  dependencies:
    bar.org: ^2

development:
  dependencies:
   baz.com: ^3
```

Which seems to allow for everything.

Multifiles is a possibility, but it always seemed messy to me.
